airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -0,0 +1,839 @@
+            raise e
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json_response = response.json()
+        if self.data_field is not None:
+            data = json_response.get(self.data_field, [])
+        elif isinstance(json_response, list):
+            data = json_response
+        elif isinstance(json_response, dict):
+            data = [json_response]
+
+        for record in data:
+            yield record
+
+        # wait for 9 seconds according to API limit
+        time.sleep(3600 / self.rate_limit)
+
+
+class IncrementalMixpanelStream(MixpanelStream, ABC):
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
+        current_stream_state = current_stream_state or {}
+        current_stream_state: str = current_stream_state.get("date", str(self.start_date))
+        latest_record_date: str = latest_record.get(self.cursor_field, str(self.start_date))
+        return {"date": max(current_stream_state, latest_record_date)}
+
+
+class Cohorts(MixpanelStream):
+    """Returns all of the cohorts in a given project.
+    API Docs: https://developer.mixpanel.com/reference/cohorts
+    Endpoint: https://mixpanel.com/api/2.0/cohorts/list
+
+    [{
+        "count": 150
+        "is_visible": 1
+        "description": "This cohort is visible, has an id = 1000, and currently has 150 users."
+        "created": "2019-03-19 23:49:51"
+        "project_id": 1
+        "id": 1000
+        "name": "Cohort One"
+    },
+    {
+        "count": 25
+        "is_visible": 0
+        "description": "This cohort isn't visible, has an id = 2000, and currently has 25 users."
+        "created": "2019-04-02 23:22:01"
+        "project_id": 1
+        "id": 2000
+        "name": "Cohort Two"
+    }
+    ]
+
+    """
+
+    data_field = None
+    primary_key = "id"
+
+    def path(self, **kwargs) -> str:
+        return "cohorts/list"
+
+
+class FunnelsList(MixpanelStream):
+    """List all funnels
+    API Docs: https://developer.mixpanel.com/reference/funnels#funnels-list-saved
+    Endpoint: https://mixpanel.com/api/2.0/funnels/list
+    """
+
+    primary_key = "funnel_id"
+    data_field = None
+
+    def path(self, **kwargs) -> str:
+        return "funnels/list"
+
+
+class DateSlicesMixin:
+    def stream_slices(
+        self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        date_slices = []
+
+        # use the latest date between self.start_date and stream_state
+        start_date = self.start_date
+        if stream_state:
+            # Remove time part from state because API accept 'from_date' param in date format only ('YYYY-MM-DD')
+            # It also means that sync returns duplicated entries for the date from the state (date range is inclusive)
```

Comment:
can we also make sure to note this in the connector docs page?

Reply:
added now to docs page